apps/desktop/src/lib/commit/StackingUpstreamCommitsAccordion.svelte

```diff
@@ -0,0 +1,116 @@
+<script lang="ts">
+	import Icon from '@gitbutler/ui/Icon.svelte';
+	import type { Snippet } from 'svelte';
+
+	interface Props {
+		children: Snippet;
+		action: Snippet;
+		remoteCommitCount: number;
+	}
+
+	let isOpen = $state(false);
+
+	const { children, action, remoteCommitCount }: Props = $props();
+
+	function toggle() {
+		isOpen = !isOpen;
+	}
+</script>
+
+<button class="accordion" onclick={toggle}>
+	<div class="accordion-row">
+		<div class="accordion-row__line dots">
+			{#each new Array(remoteCommitCount) as _, idx}
+				<svg
+					width="14"
+					height="14"
+					viewBox="0 0 14 14"
+					class="upstream-dot"
+					style="--dot: {idx + 1}; --dotCount: {remoteCommitCount + 1};"
+					fill="none"
+					xmlns="http://www.w3.org/2000/svg"
+				>
+					<rect x="1.76782" y="1.76764" width="10.535" height="10.535" rx="3" stroke-width="2" />
+				</svg>
+			{/each}
+		</div>
+		<div class="accordion-row__right">
+			<h5 class="text-13 text-body text-semibold title">Upstream commits</h5>
+			<Icon name={isOpen ? 'chevron-down' : 'chevron-up'} />
+		</div>
+	</div>
+	{#if isOpen}
+		<div class="accordion-row">
+			{@render children()}
+		</div>
+
+		<div class="accordion-row unthemed">
+			<div class="accordion-row__line"></div>
+			<div class="accordion-row__right">
+				{@render action()}
+			</div>
+		</div>
+	{/if}
+</button>
+
+<style>
+	.accordion {
+		position: relative;
+		display: flex;
+		flex-direction: column;
+		background-color: var(--clr-theme-warn-bg);
+
+		& .accordion-row {
+			display: flex;
+			width: 100%;
+			min-height: 44px;
+			align-items: stretch;
+			text-align: left;
+			border-bottom: 1px solid var(--clr-border-2);
+
+			&.unthemed {
+				background-color: var(--clr-bg-1);
+			}
+
+			&:not(:last-child) {
+				border-bottom: 1px solid var(--clr-border-2);
+			}
+
+			& .accordion-row__line {
+				margin: 0 10px 0 22.5px;
+				border-right: 2px solid var(--clr-commit-upstream);
+
+				&.dots {
+					place-items: center;
+				}
+
+				& .upstream-dot {
+					position: absolute;
+					fill: var(--clr-commit-upstream);
+					stroke: var(--clr-core-warn-80);
+					transform: translateX(-6px) translateY(calc(var(--dot) * 7.5px)) rotate(45deg);
+					/* transform: translateX(-6px) translateY(calc(22px / var(--dotCount, 1))) rotate(45deg); */
+				}
+			}
+
+			& .accordion-row__right {
+				display: flex;
+				flex: 1;
+				margin: 0 8px;
+				align-items: center;
+				color: var(--clr-text-2);
+			}
+
+			& .title {
+				flex: 1;
+				display: block;
+				color: var(--clr-text-1);
+				width: 100%;
+			}
+		}
+
+		&:focus {
+			outline: none;
+		}
+	}
```

Comment:
I feel like we could flatten this a bit since component css is isolated and won't interfere with other components. In other words, I don't think we should nest `.accordion-row` under `.accordion`.

Reply:
Makes sense :+1: